xray/test/test_dataset.py

```diff
@@ -1518,9 +1531,9 @@ def test_to_and_from_dataframe(self):
         self.assertDatasetIdentical(ds, Dataset.from_dataframe(actual))
 
         # check pathological cases
-        df = pd.DataFrame([1])
+        df = pd.DataFrame([1], columns=['A'])
```

Comment:
We raise a type error. The alternative is to cast the integers to strings, which won't roundtrip.